mailer/engine.py

```diff
@@ -112,6 +113,25 @@ def release_lock(lock):
     logging.debug("released.")
 
 
+def handle_backend_exception(connection, message, err):
+    """ simple error handler that marks messages failing due to SMTP
+    issues as deferred.  Only need to return the action type """
+    error_type = type(err)
+    action = None
+    if isinstance(error_type, (socket_error, smtplib.SMTPSenderRefused,
```

Comment:
`ìsinstance` is used on a class/type rather than an instance. That looks like a bug.

Reply:
Actually I believe we needed to convert it to a class/type in order to catch it.  Which is why the original pull request was the way it was.

Reply:
I verified now that use of `type` is a problem to be sure. See:
```
In [4]: from socket import error as socket_error
   ...: try:
   ...:     raise socket_error
   ...: except Exception as err:
   ...:     error_type = type(err)
   ...: isinstance(error_type, socket_error)
   ...: 
Out[4]: False

In [5]: from socket import error as socket_error
   ...: try:
   ...:     raise socket_error
   ...: except Exception as err:
   ...:     pass
   ...: isinstance(err, socket_error)
   ...: 
Out[5]: True
```